sky/shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -132,94 +134,144 @@ public AccessibilityNodeInfo createAccessibilityNodeInfo(int virtualViewId) {
-            if (newNode != null) {
-                sendAccessibilityEvent(newNode.id, AccessibilityEvent.TYPE_VIEW_HOVER_ENTER);
+            if (newObject != null) {
+                sendAccessibilityEvent(newObject.id, AccessibilityEvent.TYPE_VIEW_HOVER_ENTER);
             }
-            if (mHoveredNode != null) {
-                sendAccessibilityEvent(mHoveredNode.id, AccessibilityEvent.TYPE_VIEW_HOVER_EXIT);
+            if (mHoveredObject != null) {
+                sendAccessibilityEvent(mHoveredObject.id, AccessibilityEvent.TYPE_VIEW_HOVER_EXIT);
             }
-            mHoveredNode = newNode;
+            mHoveredObject = newObject;
         }
     }
 
     @Override
     public void updateSemanticsTree(SemanticsNode[] nodes) {
+        Set<SemanticObject> updatedObjects = new HashSet<SemanticObject>();
+        Set<SemanticObject> removedObjects = new HashSet<SemanticObject>();
         for (SemanticsNode node : nodes) {
-            updateSemanticsNode(node);
+            updateSemanticObject(node, updatedObjects, removedObjects);
             sendAccessibilityEvent(node.id, AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED);
         }
+        for (SemanticObject object : removedObjects) {
+            if (!updatedObjects.contains(object)) {
+                removeSemanticObject(object, updatedObjects);
+            }
+        }
+    }
+
+    private SemanticObject updateSemanticObject(SemanticsNode node,
+                                                Set<SemanticObject> updatedObjects,
+                                                Set<SemanticObject> removedObjects) {
+        SemanticObject object = mObjects.get(node.id);
+        if (object == null) {
+            object = new SemanticObject();
+            mObjects.put(node.id, object);
+        }
+        object.updateWith(node);
+        updatedObjects.add(object);
+        if (node.children != null) {
+            if (node.children.length == 0) {
+                if (object.children != null) {
+                    removedObjects.addAll(object.children);
+                }
+                object.children = null;
+            } else {
+                if (object.children == null) {
+                    object.children = new ArrayList<SemanticObject>(node.children.length);
+                } else {
+                    removedObjects.addAll(object.children);
+                    object.children.clear();
+                }
+                for (SemanticsNode childNode : node.children) {
+                    SemanticObject childObject = updateSemanticObject(childNode, updatedObjects, removedObjects);
+                    childObject.parent = object;
+                    object.children.add(childObject);
+                }
+            }
+        }
+        if (node.geometry != null) {
+            // has to be done after children are updated
+            // since they also get marked dirty
+            object.invalidateGlobalGeometry();
```

Comment:
It just means that this function walks the children so needs to be called after the children are updated.